op-batcher/batcher/driver.go

```diff
@@ -606,3 +678,50 @@ func logFields(xs ...any) (fs []any) {
 	}
 	return fs
 }
+
+// JamTxPool is for testing ONLY. It sends a txpool-blocking transaction in a new (non-blocking)
+// goroutine to the txpool that will panic if the transaction completes successfully. (Tests should
+// expect the blocking transaction to result in error from being cancelled.) This function must be
+// called *before* the batcher starts submitting batches to ensure successful jamming, and will
+// panic otherwise.
+func (l *BatchSubmitter) JamTxPool(ctx context.Context) {
```

Comment:
Could `JamTxPool` call an internally-held `jamTxPoolFunc` that is stored in the struct? That way, production BatchSubmitters would have no way of triggering panics, while test-BatchSubmitters could be given this testing behavior.

Reply:
Not sure I follow, though I'm all ears if there's a clever way to prevent this from being used accidentally in prod.

Reply:
I mean that the Batch Submitter struct could contain,

```go
type BatchSubmitter struct {
...
jamPoolFn    func(context.Context)
...
}
```
And then this function you have here would proxy through to the stored function,
```go
func (l *BatchSubmitter) JamTxPool(ctx context.Context){
    if l.jamPoolFn != nil {
        l.jamPoolFn(ctx)
    }
    // implied else: no function set, nothing to do
}
```

This way, even calling this test function won't have any production effects. You'd set the function on the Batch Submitter in the tests that use it.

Reply:
Ah I see. The jamming function needs to access the internal / non-exported methods of BatchSubmitter though so the func couldn't be defined outside the package. So not sure that buys us that much?

Reply:
@sebastianst nice idea, done!